### PR TITLE
Add btrfs support to image customization and creation

### PR DESCRIPTION
**Pull request by vinceaperri**
Adds support for the btrfs filesystem type when creating and customizing Azure Linux images.

- Add btrfs as a valid partition type in the API
- Add btrfs support to partition formatting
  - Ensures filesystem features are consistent across mkfs.btrfs version
  - Notably lacks subvolume support
- Add btrfs support to SELinux file relabeling
- Add btrfs support when checking filesystems after image customization
- Add btrfs support when resetPartitionsUuidsType is `reset-all`
- Fix rootfs search ending prematurely for btrfs partitions, handling multiple fstabs edge-case.
- Update prerequisites in docs to include mkfs.btrfs and btrfs-progs package

---

### **Checklist**
- [x] Tests added/updated
- [x] Documentation updated (if needed)
- [x] Code conforms to style guidelines

### CODE REVIEW

File: toolkit/tools/pkg/imagecustomizerlib/partitionutils_test.go [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+		return
+	}
+
+	mountDir := filepath.Join(testTmpDir, "mount")
+	err = os.MkdirAll(mountDir, 0755)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	btrfsMount, err := safemount.NewMount(loopback.DevicePath(), mountDir, "btrfs", 0, "", false)
```
Comment (liulanze) [inline toolkit/tools/pkg/imagecustomizerlib/partitionutils_test.go]: Missing `defer btrfsMount.Close()` and `CleanClose()` potentially.

**Review by cwize1 — CHANGES_REQUESTED** [review]

**Review by cwize1 — CHANGES_REQUESTED** [review]

File: toolkit/tools/imagegen/diskutils/filesystem.go [hunk 2]
```diff
@@ -99,22 +126,39 @@ var (
...
 			Ext4:    fedora42Ext4Options,
 			Xfs:     fedora42XfsOptions,
 			BootXfs: fedora42XfsOptions,
 		},
 	}
 
+	// A list of btrfs features and their minimum supported kernel versions.
+	//
+	// Note: This list omits features that either:
+	// - Are not used by one of the supported distros/versions, OR
```
[the remark this answers is not in the document]
Comment (cwize1) [reply toolkit/tools/imagegen/diskutils/filesystem.go]: I meant in `btrfsFeaturesKernelSupport`. That way, you don't need a complex comment describing why some features aren't mentioned.
